From 4bd0d8c22ca65ba2bd02c671048a99e414f049b8 Mon Sep 17 00:00:00 2001
From: Ryan Schmidt <git@ryandesign.com>
Date: Wed, 23 Oct 2019 16:31:55 -0500
Subject: [PATCH] build: don't use -latomic on macOS

Fixes build when using open-source clang.

Builds using Apple clang from Xcode v7 and later were already working,
due to the fact that node's build system misidentifies their
llvm_version as "0.0" and thus the flag wasn't being added.

PR-URL: https://github.com/nodejs/node/pull/30099
Fixes: https://github.com/nodejs/node/issues/30093
Reviewed-By: Richard Lau <riclau@uk.ibm.com>
Reviewed-By: Ruben Bridgewater <ruben@bridgewater.de>
---
 node.gyp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/node.gyp b/node.gyp
index 781dcee81adb..9389aa424640 100644
--- a/node.gyp
+++ b/node.gyp
@@ -298,7 +298,7 @@
           '-Wl,-bnoerrmsg',
         ],
       }],
-      ['OS in ("linux", "mac") and llvm_version != "0.0"', {
+      ['OS == "linux" and llvm_version != "0.0"', {
         'libraries': ['-latomic'],
       }],
     ],
